### ISSUE TITLE
Support the Builder pattern

### ISSUE THREAD

**Issue by Kromey** [Kromey/fast_poisson#2]
Should `fast_poisson` support the Builder pattern? Usage could then look like this:
```rust
let poisson = Poisson::new().size(400, 400).radius(5);
```
Compare to existing code needed to do the same thing:
```rust
let poisson = Poisson {
    width: 400,
    height: 400,
    radius: 5,
    ..Default::default()
};
```
Both methods work just fine, but some may prefer the ergonomics of the builder pattern.

**Comment by Kromey** [comment]
This will be implemented so that internal representation can be hidden, such as our change from `seed` being a `u64` to being `Option<u64>`.

**Comment by Kromey** [comment]
Implemented in 9fab0594ee006c5115a5a58063bc6ad4f0e737fd
